app/admin/case_document.rb

```diff
@@ -1,9 +1,63 @@
-ActiveAdmin.register Case do
+require "mini_magick"
+
+ActiveAdmin.register CaseDocument do
   menu parent: "Cases", priority: 1
 
+  form title: "Document upload" do |f|
+    f.inputs "document" do
+      f.input :case
+      f.input :title
+      f.input :document, as: :file
+      f.label "convert to: "
+      f.select :document_type, %w[png jpg tiff]
+    end
+    f.actions
+  end
+
   controller do
     def permitted_params
       params.permit!
     end
+
+    def create
+      case_document = CaseDocument.new(permitted_params[:case_document])
+      case_document.document.attach(permitted_params[:case_document][:document])
+      case_document.save!
+      # The following will be done in Active Job for async
+      # Assuming an image is uploaded, not any other file type
+      transcode_image(case_document, case_document.document_type)
```

Comment:
We can run this from the ConvertImageToPngJob, so the controller doesn't wait until this operation complete